examples/tests/test_examples.py

```diff
@@ -22,38 +24,56 @@
 
 
 # run all tests unless they opt-out
-examples_to_run = find_examples(negative_query="# run_example = false")
+examples_to_run = find_examples(
+    negative_query="# run_example = false", return_stems=True
+)
 
 # only test output of examples that opt-in
 examples_to_test = find_examples(query="# test_example = true", return_stems=True)
 
 
-@pytest.mark.parametrize("module", examples_to_run, ids=lambda module: module.stem)
-def test_examples_run(module):
+@pytest.mark.parametrize("module", examples_to_run)
+def test_examples_run(module, force_offscreen, disable_call_later_after_run):
     """Run every example marked to see if they can run without error."""
-    env = os.environ.copy()
-    env["WGPU_FORCE_OFFSCREEN"] = "true"
+    # use runpy so the module is not actually imported (and can be gc'd)
+    # but also to be able to run the code in the __main__ block
+    runpy.run_module(f"examples.{module}", run_name="__main__")
 
+
+@pytest.fixture
+def disable_call_later_after_run():
+    """Disable call_later after run has been called."""
+    # we start by asserting no tasks are pending
+    # if this fails, we likely need to refactor this fixture
+    loop = asyncio.get_event_loop_policy().get_event_loop()
+    if len(asyncio.all_tasks(loop=loop)) != 0:
+        raise RuntimeError("no tasks should be pending")
+
+    orig_run = wgpu.gui.offscreen.run
+    orig_call_later = wgpu.gui.offscreen.call_later
+    allow_call_later = True
+
+    def wrapped_call_later(*args, **kwargs):
+        if allow_call_later:
+            orig_call_later(*args, **kwargs)
+
+    def wrapped_run(*args, **kwargs):
+        nonlocal allow_call_later
+        allow_call_later = False
+        orig_run(*args, **kwargs)
+
+    wgpu.gui.offscreen.call_later = wrapped_call_later
+    wgpu.gui.offscreen.run = wrapped_run
```

Comment:
This stuff can be removed once there is a new release of wgpu-py that has the upstream fix. I will make an issue to track that.

Reply:
#405